riscv/src/compiler.rs

```diff
@@ -671,6 +671,8 @@ fn mul_instruction<T: FieldElement>() -> &'static str {
 "#
         }
         KnownField::GoldilocksField => {
+            // The Goldilocks field cannot fit any 64-bit number, so we have to use
```

Comment:
any -> some

I still don't fully understand. If it can fit the product, why do we need the split machine? Are the inputs not properly constrained?

Reply:
If you look at the old implementation (code still above) that works for BN, this is what the prover could overflow: `X + Y * 2**32`

Reply:
ok, understood!